@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -119,13 +120,12 @@ class PoiRenderBufferBatch {
             trilinear ? THREE.LinearMipMapLinearFilter : THREE.NearestFilter,
             THREE.RGBAFormat
         );
-        texture.needsUpdate = true;
-        texture.premultiplyAlpha = premultipliedAlpha;
         // Generate mipmaps for distance scaling of icon
-        // TODO: Implement custom mip map generation if support for texture atlas
-        // with icons that are not power of two or have different sizes is necessary.
-        texture.generateMipmaps = true;
+        texture.mipmaps = mipMapGenerator.generateTextureAtlasMipMap(iconTexture.image);
```

Comment:
I get the feeling this should all be optional. I.e. for plain top-down usecases you don't want to waste GPU resources and processing for the mip map generation. Especially b/c it's done in the main thread. Do you have any performance numbers comparing POIs w/o mipmaps vs. POIs w/ mipmaps. 
I know it's hard to get numbers with our perf. tests but maybe you can try locally using the benchmark example

Reply:
I agree, however do we have a means to enforce such a use case? If not, I think this is fine until such an API exists to have a top down use case, however.... if that is the use case, there are other map rendering engines which are much better in the 2D case, so it probably isn't worth doing.

Reply:
Even for top down views trilinear filtering/mipmaps can improve the image quality of the icons. The map theme allows to freely scale the size of the icon which can lead to some artifacts with just bilinear filtering enabled. And the memory overhead of mipmaps isn't that big only 1/3 more memory if the texture is already power-of-two. Slightly more if padding is added. Additionally, the current themes only use two relatively small textures for icons so the overall memory overhead should just be 1-2 MB in GPU memory. I will try locally how big the performance impact/memory overhead is.

Reply:
Just running the numbers of paper:
|Texture|size without mipmaps|size with mipmaps and padding|
| ------------- | ------------- | ------------- |
|icons_day.png|329×296×4 Byte = 389,54 kB|512x512x1,3333x4 Byte = 1,4 MB|
|city_center.png|38x38x4 Byte = 5,78 kB|64x64x1,3333x4 Byte = 21,84 kB|
|**Total**|395,32 kB|1,421 MB|

Reply:
I added links to the benchmarks in the JIRA ticket. Overall only very minor differences when it comes to performance and memory consumption.